### ISSUE TITLE
No se ve la epg de los 7 días 7UD.

### ISSUE THREAD

**Issue by joseluispolar** [Paco8/plugin.video.movistarplus#9]
Buenas Paco,

Buen trabajo.
Comentarte qyue he cargado la versión 0.75 en kodi 20.2 y al sacar o exportar guia y ver los 7 ultimos días, sólo me aparece el día anterior y algun programa de 2 días antes. Sin embaego desde la app de Movistar si veo hasta los 7 días anteriores.  En.la configuración de Kodi le tengo pueato 8 días antes. ¿Podria ser algún problema de la versión de tu addons?

Saludos,

**Comment by Paco8** [comment]
Sí, es un fallo del addon. Veré si puedo arreglarlo.

**Comment by joseluispolar** [comment]
Si no dime por que parte esta de los modulos, por lo mirarme todo y lo
corrijo yo.

Gracias

**Comment by Paco8** [comment]
Más que un fallo es que actualmente solo se descargan dos días.
Si quieres mirarlo, en la función [get_epg](https://github.com/Paco8/plugin.video.movistarplus/blob/main/resources/lib/movistar.py#L418) habría que cambiar la fecha de inicio (a 7 días atrás) y cambiar el DURATION a 7.

**Comment by joseluispolar** [comment]
Voy a cambiarlo y ver si funciona correctamente y te digo.
También he detectado algún error que da en el registro al traer
el.manifest.mdf viendo el canal, que da 200 en la respuesta pero el.log
dice que error, lo.miraré a ver si es un error o que

Gracias,

**Comment by macamba** [comment]
Buenas,

He intentado cambiar el valor de DURATION a 7 en la funcion get_epg pero me da error  También probe con 3 y 4 y no dio error pero no surtio efecto 😢

Saludos

**Comment by Paco8** [comment]
Sí, parece que si se intentan descargar más de 3 días da error (ERROR 502: Bad Gateway).

**Comment by macamba** [comment]
Hola Paco,

Parece que ahora si que deja poner mas dias cambiando DURATION de valor pero tiene un problema, y es que solo van apareciendo mas dias en el futuro no en el pasado. Lo mas viejo q se puede ver es el ultimo programa que emitieron el dia anterior :(

**Comment by Paco8** [comment]
Tienes que cambiar la fecha inicial.

```
+        from datetime import timedelta
         today = datetime.today()
+        today -= timedelta(days=7)
         str_now = today.strftime('%Y-%m-%dT00:00:00')

```
De todas maneras a mí me sigue sin cargar la epg si le pongo un DURATION de más de 3 días.

**Comment by macamba** [comment]
Si, he probado pero da igual lo q pongas, solo se ven 3 días

**Comment by joseluispolar** [comment]
Si tal cual tienen la aplicación, la carga de la EPG es por días, según
pasas con el.mando. Quizás la forma desdd kody sería implementar eso, una
carga tipo como la de la app de movistar

**Comment by Paco8** [comment]
Eso había pensado yo también. Lo haré cuando pueda.

**Comment by joseluispolar** [comment]
Contad conmigo !!! Os puedo ayudar en el.desarrollo.
